rust/src/ike/detect.rs

```diff
@@ -0,0 +1,263 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// Author: Frank Honza <frank.honza@dcso.de>
+
+use crate::ike::ike::*;
+use std::ptr;
+use std::ffi::CStr;
+use super::ipsec_parser::IkeV2Transform;
+
+#[no_mangle]
+pub extern "C" fn rs_ike_state_get_exch_type(
+    tx: &mut IKETransaction,
+    exch_type: *mut u8,
+) -> u8 {
+    if exch_type == std::ptr::null_mut() {
+        return 0;
+    }
+
+    if tx.ike_version == 1 {
```

Comment:
style: I wonder if we should rather use `match` than `if else if`

Reply:
hmmm first i had it with a match-statement, but with the obligatory "_" it had 3 options and found the if/else way somehow nicer...no strong opinion here....

Reply:
No strong opinion neither...

Reply:
then i'll leave it that way :)